File: src/main/java/uk/co/threebugs/mochiwhattotrade3/account/AccountService.java
package uk.co.threebugs.mochiwhattotrade3.account;

import lombok.AllArgsConstructor;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.Optional;

@Service
@AllArgsConstructor
public class AccountService {

    private final AccountRepository accountRepository;
    private final AccountMapper accountMapper;

    public Optional<AccountDto> findByName(String name) {
        return accountRepository.findByName(name)
                                .map(accountMapper::toDto);
    }

    public AccountDto save(AccountDto accountDto) {
        return accountMapper.toDto(accountRepository.save(accountMapper.toEntity(accountDto)));
    }

    public List<AccountDto> findAll() {
        return accountRepository.findAll()
                                .stream()
                                .map(accountMapper::toDto)
                                .toList();
    }
}

File: src/main/kotlin/uk/co/threebugs/mochiwhattotrade3/account/AccountService.kt
package uk.co.threebugs.mochiwhattotrade3.account

import org.springframework.stereotype.Service

@Service
class AccountService(
        private val accountRepository: AccountRepository,
        private val accountMapper: AccountMapper
) {
    fun findByName(name: String): AccountDto? {
        val account = accountRepository.findByName(name).orElse(null)
        return account?.let { accountMapper.toDto(it) }
    }

    fun save(accountDto: AccountDto): AccountDto {
        val savedAccount = accountRepository.save(accountMapper.toEntity(accountDto))
        return accountMapper.toDto(savedAccount)
    }

    fun findAll(): List<AccountDto> {
        return accountRepository.findAll()
                .map { accountMapper.toDto(it) }
    }
}
